### PR TITLE
Replace charts with react components

### LINKED ISSUE #20: replace charts with react components
The current chart solution is deprecated and needs to be replaced by react components.

### PR DESCRIPTION
**Pull request by kamils-iRonin**
This closes #20 
image

### CODE REVIEW

File: test/test_plugin_helper.rb [hunk 1]
```diff
@@ -20,3 +20,17 @@ class Spec
     end
   end
 end
+
+module ForemanOmaha
```
Comment (timogoebel) [inline test/test_plugin_helper.rb]: Does it make sense to have this in core as well?

Comment (kamils-iRonin) [reply test/test_plugin_helper.rb]: It makes sense, there is a problem with test when we use `webpacked_plugins_js_for` and we try to register fake `:skip_test` plugin. This is to make possible to register `:skip_test` plugin.
I've checked and when https://github.com/theforeman/foreman/blob/develop/test/test_helper.rb#L30-L32 is commented out then the test passes as well.

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{
+  "root": true,
```
[the remark this answers is not in the document]
Comment (sharvit) [reply .eslintrc]: I would strongly recommend using the same configurations as in the core using [eslint-plugin-patternfly-react](https://github.com/patternfly/patternfly-react/tree/master/packages/eslint-plugin-patternfly-react).

See:
https://github.com/theforeman/foreman/commit/839bcbbd1499e6dd377ef8cf7f4ba67742c55eed

**Comment by ohadlevy** [comment]
can this somehow be added to the plugin template at
https://github.com/theforeman/foreman_plugin_template ?

File: test/test_plugin_helper.rb (hunk 1 above)
Comment (ekohl) [reply test/test_plugin_helper.rb]: We have https://github.com/theforeman/foreman/blob/1d28598ff33ed287d192754deaee988508e18ea0/app/registries/foreman/plugin/assets.rb#L33-L35 so I wonder if that could be used

**Comment by timogoebel** [comment]
As discussed with @ekohl on IRC, I'd wait for the packaging issues to be solved before merging this PR.

File: .eslintrc (hunk 2 above)
[the remark this answers is not in the document]
Comment (sharvit) [reply .eslintrc]: Now you can add `.travis.yml` if you want to have a CI that catches linting issues.
Can also add [jest](https://jestjs.io/) with some tests and run them in Travis.

[the remark this answers is not in the document]
Comment (timogoebel) [reply .eslintrc]: @kamils-iRonin: Travis should now be active. Can you push again?

**Comment by timogoebel** [comment]
[test foreman_omaha]

**Comment by ezr-ondrej** [comment]
I believe now it should be much easier to finish this.

File: webpack/components/OmahaDonutChart.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { map, get } from 'lodash';
+// eslint-disable-next-line import/no-extraneous-dependencies, import/no-unresolved, import/extensions
+import DonutChart from 'foremanReact/components/common/charts/DonutChart';
+
+const OmahaDonutChart = ({ data: { columns, search } }) => {
```
Comment (ezr-ondrej) [inline webpack/components/OmahaDonutChart.js:7]: This should not be necessary, we have search on the DonutChart already.

**Comment by kamils-iRonin** [comment]
closed in favor of https://github.com/theforeman/foreman_omaha/pull/55

**Pull request closed without merging**